src/legacy/ui/public/new_platform/index.test.mocks.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export const mockNewPlatformBackdoor = () => {
+  return {
+    __setup__: () => {},
+    __start__: () => {},
+    npSetup: {
+      core: {},
+      plugins: {
+        data: {
+          interpreter: {
```

Comment:
we are adding information specific to data plugin to new_platform plugin ?

Reply:
`new_platform` is not a plugin, it is a backdoor for legacy code into NP. This adds only mocks for testing into this shared folder so every Jest test in legacy world can just use this mock, instead of re-implementing it.